cli/loader_lib.c

```diff
@@ -353,6 +353,17 @@ static char *libstdcxxprobe(void)
 void *libjulia_internal = NULL;
 void *libjulia_codegen = NULL;
 __attribute__((constructor)) void jl_load_libjulia_internal(void) {
+#if defined(_OS_LINUX_)
```

Comment:
@staticfloat I've added you as a reviewer because of this change to the cli loader.

Let me know if you'd like this done in a different way :+1:

Reply:
To be explicit, I think setting the signals to `SIG_BLOCK` without having a `sigwait()` handler running yet is okay because we only have two choices at this point:

* Execution continues to the point that we start up our signal handler, at which point any queued signals will be processed.
* Execution halts before then due to a fatal error, at which point signal handling is moot.

Reply:
I thought of one defect: This was leaving more signals blocked than necessary, including signals that we don't sigwait() on which Julia code or an application embedding Julia may wish to use. I've pushed a fix to restore the signal mask at the end of this constructor, which should fix that problem.

P.S. This also made me realize that for Julia-in-Julia, our signal handling is probably totally non-deterministic about which Julia runtime receives a signal.

Reply:
> This also made me realize that for Julia-in-Julia, our signal handling is probably totally non-deterministic about which Julia runtime receives a signal.

We could define an environment variable that disables signal handling in child Julia's.

Reply:
How about jl_init_options instead? I can spin up a PR for that on Monday